Automattic/jetpack
# Update to pnpm v9

## Proposed changes:

It's been out for a while, and people are starting to get confused again when installing the latest version doesn't work.

Additional changes:
* Drop pnpmfile hack preventing webpack without webpack-cli, the issue it was avoiding has been fixed.
* New lock file format.
* No need for hacks to get the lock file updated after editing pnpmfile anymore.

### Other information:


## Jetpack product discussion


p1717594985376749/1717594306.621839-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?


no

## Testing instructions:






* CI happy?
* Stuff still work?
  * Use `jetpack generate` to create a js-package, make sure package.json has correct versions for deps.
  * Try updating a dep on `@wordpress/dependency-extraction-webpack-plugin` to v6. It should be rejected.
